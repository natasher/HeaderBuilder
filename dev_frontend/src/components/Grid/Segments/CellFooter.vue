<template>
  <footer
    class="hbr-footer">
    <span :class="classObject">

      <!-- @slot Place <grid-row-cell /> here -->
      <slot />

    </span>
  </footer>
</template>

<script>
export default {

  name: 'CellFooter',

  props: {
    /**
     * [ 'left', 'center', 'right' ]
     */
    pos: {
      type    : String,
      required: true,
    },
  },

  computed: {
    classObject: function () {
      return `hbr-title-${ this.pos }`
    },
  },

}
</script>
