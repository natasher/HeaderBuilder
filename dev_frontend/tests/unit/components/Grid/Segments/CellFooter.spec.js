import { shallowMount } from '@vue/test-utils'
import CellFooter from '../../../../../src/components/Grid/Segments/CellFooter.vue'

describe('CellFooter.vue', () => {

  describe('snapshots:', () => {

    test('should match if position is `test` and slot is empty', () => {
      const wrapper = shallowMount(
        CellFooter,
        {
          propsData: {
            pos: 'test'
          }
        }
      )

      expect( wrapper.element ).toMatchSnapshot()
    })

    test('should match if position is `test` and slot contain some content', () => {
      const wrapper = shallowMount(
        CellFooter,
        {
          propsData: {
            pos: 'test'
          },
          slots: {
            default: '<div>DEFAULT SLOT</div>'
          }
        },
      )

      expect( wrapper.element ).toMatchSnapshot()
    })

  })

  test('should create class for footer from `pos` prop and given string', () => {
    const wrapper = shallowMount(
      CellFooter,
      {
        propsData: {
          pos: 'test',
        }
      }
    )
    const el = wrapper.find('span')

    expect(el.classes()).toEqual([ 'hbr-title-test' ])
  })

  test('should accept default slot content', () => {
    const wrapper = shallowMount(
      CellFooter,
      {
        propsData: {
          pos: 'test'
        },
        slots: {
          default: '<div>DEFAULT SLOT</div>'
        }
      }
    )

    expect(wrapper.find('div').text()).toBe('DEFAULT SLOT')
  })

})