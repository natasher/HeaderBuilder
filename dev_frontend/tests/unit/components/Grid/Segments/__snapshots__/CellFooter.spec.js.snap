// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellFooter.vue snapshots: should match if position is \`test\` and slot contain some content 1`] = `
<footer
  class="hbr-footer"
>
  <span
    class="hbr-title-test"
  >
    <div>
      DEFAULT SLOT
    </div>
  </span>
</footer>
`;

exports[`CellFooter.vue snapshots: should match if position is \`test\` and slot is empty 1`] = `
<footer
  class="hbr-footer"
>
  <span
    class="hbr-title-test"
  />
</footer>
`;
